pkg/controllers/cluster/import.go

```diff
@@ -139,6 +138,22 @@ func (i *importHandler) deleteOldAgent(cluster *fleet.Cluster, kc kubernetes.Int
 		}
 	}
 
+	if i.systemNamespace != config.DefaultNamespace {
+		if _, err := kc.CoreV1().Namespaces().Get(i.ctx, config.DefaultNamespace, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		if err := kc.AppsV1().Deployments(config.DefaultNamespace).Delete(i.ctx, config.AgentConfigName, metav1.DeleteOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		logrus.Infof("Deleted old agent for cluster (%s/%s) in namespace: %s", cluster.Namespace, cluster.Name, config.DefaultNamespace)
```

Comment:
@StrongMonkey I returned `nil` since I only want to log the successful deletion if it occurs without error (even "NotFound")